python/ray/rllib/ddpg/tests/test_graph.py

```diff
@@ -0,0 +1,138 @@
+                actions = np.asarray([elem[1] for elem in batch])
+                rewards = np.asarray([elem[2] for elem in batch])
+                obses_tp1 = np.asarray([elem[3] for elem in batch])
+                dones = np.asarray([elem[4] for elem in batch])
+                weights = np.ones_like(rewards)
+                batch_indexes = - np.ones_like(rewards)
+                samples = SampleBatch({
+                    "obs": obses_t, "actions": actions, "rewards": rewards,
+                    "new_obs": obses_tp1, "dones": dones, "weights": weights,
+                    "batch_indexes": batch_indexes})
+                '''
+                '''
+                (obses_t, actions, rewards, obses_tp1,
+                    dones) = memory.sample(
+                        DEFAULT_CONFIG["train_batch_size"])
+                weights = np.ones_like(rewards)
+                batch_indexes = - np.ones_like(rewards)
+                samples = SampleBatch({
+                    "obs": obses_t, "actions": actions, "rewards": rewards,
+                    "new_obs": obses_tp1, "dones": dones, "weights": weights,
+                    "batch_indexes": batch_indexes})
+                '''
+                (obses_t, actions, rewards, obses_tp1,
+                    dones, weights, batch_indexes) = memory.sample(
+                        DEFAULT_CONFIG["train_batch_size"], 0.4)
+                samples = SampleBatch({
+                    "obs": obses_t, "actions": actions, "rewards": rewards,
+                    "new_obs": obses_tp1, "dones": dones, "weights": weights,
+                    "batch_indexes": batch_indexes})
+                '''
+                print(samples["obs"])
+                print(samples["actions"])
+                print(samples["rewards"])
+                print(samples["new_obs"])
+                print(samples["dones"])
+                print(samples["weights"])
+                raw_input()
+                '''
+                td_error = ddpg.compute_apply(
+                    sess, samples["obs"], samples["actions"], samples["rewards"],
+                    samples["new_obs"], samples["dones"], samples["weights"])
+                new_priorities = (
+                    np.abs(td_error) + 1e-6)
+                if isinstance(memory, PrioritizedReplayBuffer):
+                    memory.update_priorities(
+                        samples["batch_indexes"], new_priorities)
+                #if num_ts - last_sync_target >= 500:
+                #    ddpg.update_target_hard(sess)
+                #    last_sync_target = num_ts
+                ddpg.update_target(sess)
+
+            ep_rwd += rew
+            if done:
+                ep_rwds.append(ep_rwd)
+                mean_rwd = np.mean(ep_rwds[-min(t, 100):])
+                print("%d %.3f %.3f %.3f" % (t, ep_rwd, eps, mean_rwd))
+                ddpg.reset_noise(sess)
+                break
+            ob = new_ob
+
+if __name__=="__main__":
+    main()
```

Comment:
Is this intended to be an automated test?

Reply:
removed.